DCSS block device driver changes:
 - Fix remove_store function, put_device is called too early.

diffstat:
 drivers/s390/block/dcssblk.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff -urN linux-2.6/drivers/s390/block/dcssblk.c linux-2.6-s390/drivers/s390/block/dcssblk.c
--- linux-2.6/drivers/s390/block/dcssblk.c	Sun Apr  4 05:36:26 2004
+++ linux-2.6-s390/drivers/s390/block/dcssblk.c	Tue Apr 13 10:20:34 2004
@@ -548,10 +548,10 @@
 	dev_info->gd->queue = NULL;
 	put_disk(dev_info->gd);
 	device_unregister(&dev_info->dev);
-	put_device(&dev_info->dev);
 	segment_unload(dev_info->segment_name);
 	PRINT_DEBUG("Segment %s unloaded successfully\n",
 			dev_info->segment_name);
+	put_device(&dev_info->dev);
 	rc = count;
 out_buf:
 	kfree(local_buf);
